sentry_sdk/integrations/bottle.py

```diff
@@ -0,0 +1,165 @@
+from __future__ import absolute_import
+
+import types
+
+from sentry_sdk.hub import Hub
+from sentry_sdk.utils import capture_internal_exceptions, event_from_exception
+from sentry_sdk.integrations import Integration
+from sentry_sdk.integrations.wsgi import SentryWsgiMiddleware
+from sentry_sdk.integrations._wsgi_common import RequestExtractor
+
+if False:
+    from sentry_sdk.integrations.wsgi import _ScopedResponse
+    from typing import Any
+    from typing import Dict
+    from typing import Callable
+    from typing import Optional
+    from bottle import FileUpload, FormsDict, LocalRequest  # type: ignore
+
+from bottle import Bottle, request as bottle_request  # type: ignore
+
+
+class BottleIntegration(Integration):
+    identifier = "bottle"
+
+    transaction_style = None
+
+    def __init__(self, transaction_style="endpoint"):
+        # type: (str) -> None
+        TRANSACTION_STYLE_VALUES = ("endpoint", "url")
+        if transaction_style not in TRANSACTION_STYLE_VALUES:
+            raise ValueError(
+                "Invalid value for transaction_style: %s (must be in %s)"
+                % (transaction_style, TRANSACTION_STYLE_VALUES)
+            )
+        self.transaction_style = transaction_style
+
+    @staticmethod
+    def setup_once():
+        # type: () -> None
+
+        old_app = Bottle.__call__
+
+        def sentry_patched_wsgi_app(self, environ, start_response):
+            # type: (Any, Dict[str, str], Callable) -> _ScopedResponse
+
+            hub = Hub.current
+            integration = hub.get_integration(BottleIntegration)
+            if integration is None:
+                return old_app(self, environ, start_response)
+
+            # monkey patch method self(Bottle).router.match -> (route, args)
+            # to monkey patch route.call
+            old_match = self.router.match
+
+            def patched_match(self, *args, **kwargs):
```

Comment:
I agree the patching done in this way is not really nice and a bit confusing. I tried to patch the other functions outside of `sentry_patched_wsgi_app` and I was more or less successful.

Just instead of `Route.call` (which is a cached_property) I  managed to patch `Router._make_callback`. The tests on my machine seems to work.